include/core/checkpoint_control.h

```diff
@@ -0,0 +1,91 @@
+// SPDX-FileCopyrightText: Copyright (c) 2020-2024 The Lethe Authors
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception OR LGPL-2.1-or-later
+
+#ifndef lethe_checkpoint_control_h
+#define lethe_checkpoint_control_h
+
+#include <boost/archive/text_iarchive.hpp>
+#include <boost/archive/text_oarchive.hpp>
+
+class CheckpointControl
+{
+public:
+  /**
+   * @brief The CheckpointControl is responsible for keeping track of the
+   * next checkpoint id to use.
+   *
+   * @param checkpoint_freq Checkpoint frequency.
+   * @param restart_filename Filename used for the prefix during checkpointing.
+   */
+  CheckpointControl(const unsigned int checkpoint_freq,
+                    const std::string &restart_filename)
+    : max_checkpoint_id(2)
+    , next_checkpoint_id(max_checkpoint_id)
+    , checkpointing_frequency(checkpoint_freq)
+    , filename(restart_filename)
```

Comment:
I'm just curious, butinstead of passing just two parameters like that, why not pass the entire structure of checkpoint parameters ?

Reply:
Passing dem_parameter as an input requires the class to be templated no?

Reply:
No I meant the structure that contains the restart parameters
in a similar way the simulation control gets the strucutre that contains all of the simulation control parameters